modin/pandas/base.py

```diff
@@ -2336,20 +2336,15 @@ def shift(self, periods=1, freq=None, axis=0, fill_value=no_default):
 
         if freq is None:
             if axis == "index" or axis == 0:
-                if periods > 0:
-                    dropped_df = self.drop(self.index[-periods:])
-                    new_frame = filled_df.append(dropped_df, ignore_index=True)
-                    new_frame.index = self.index.copy()
-                    if isinstance(self, DataFrame):
-                        new_frame.columns = self.columns.copy()
-                    return new_frame
-                else:
-                    dropped_df = self.drop(self.index[:-periods])
-                    new_frame = dropped_df.append(filled_df, ignore_index=True)
-                    new_frame.index = self.index.copy()
-                    if isinstance(self, DataFrame):
-                        new_frame.columns = self.columns.copy()
-                    return new_frame
+                new_frame = (
+                    filled_df.append(self.iloc[:-periods], ignore_index=True)
+                    if periods > 0
+                    else self.iloc[-periods:].append(filled_df, ignore_index=True)
+                )
```

Comment:
It turns out that the speedup is achieved by using `iloc` instead of `drop`?

Reply:
Yes, only `drop`->`iloc` for case `shift(axis=0)`. Moreover, I tried include this change in branch for `axis=1`, but got worse performance results.